numexpr/interpreter.cpp

```diff
@@ -1449,7 +1449,7 @@ NumExpr_run(NumExprObject *self, PyObject *args, PyObject *kwds)
     /* Get the sizes of all the operands */
     dtypes_tmp = NpyIter_GetDescrArray(iter);
     for (i = 0; i < n_inputs+1; ++i) {
-        self->memsizes[i] = dtypes_tmp[i]->elsize;
+        self->memsizes[i] = PyDataType_ELSIZE(dtypes_tmp[i]);
     }
```

Comment:
Wonder if it is possible to rewrite this with `PyArray_ITEMSIZE` as that works on NumPy 1 & 2 based on [this release note]( https://numpy.org/devdocs/release/2.0.0-notes.html#descriptor-elsize-and-alignment-access )